### ISSUE TITLE
[Beta 2.0.5] Nexus Site issues - No login, improperly handling links

### ISSUE THREAD

**Issue by Leer0x** [TanninOne/modorganizer#713]
MO does not properly login to Nexus Mods, it logs in correctly on download, but never manually: Gives error message saying password is wrong (It isn't, because it actually logs in with that password after pressing download and using the same one on the nexus site).
Nexus links won't launch at all from browser, just darkens webpage and goes back to normal, as if it was going to download, but no "Download started." message from MO. Nexus links do work with Nexus Mod Manager, however, and have always re-associated once message pops up on Mod Organizer, and through settings menu.

After reinstalling and deleting all Mod Organizer residue, problems still persist.
Restarting PC has no effect.
Turning off Anti-Virus has no effect. 
Running as admin has no effect.

**Comment by Wadu436** [comment]
what LOOT version are you using?

**Comment by Wadu436** [comment]
Also could you check the NMM version (Settings>Workarounds)

**Comment by Leer0x** [comment]
NMM Version is set to 0.61.13
Changed it to 0.61.23 and has no effect.
I'm also using LOOT Snapshot v0.8.1-323 (Latest Fallout 4 Nexus version, supports Fallout 4 update 1.5.157)
Snapshot builds I believe are the only ones that support Fallout 4

**Comment by Wadu436** [comment]
I'm using both of those as well and they're working fine for me, so I'm sorry, I can't help you on this one

**Comment by RPINerd** [comment]
LOOT was never working for me.. Every time I clicked the "Sort" button all I got was Error Code 1 and:

19:30:08 [D] [loot] Error: invalid dll, variable loot_game_fo4 not found

As far as Nexus Login, the beta seems to be working just fine for me atm

**Comment by Akzyra** [comment]
Open nxmhandler.exe and check if fallout4 is set to the right MO2 executable.
If it is, you have some other bug, downloads are working for me.

**Comment by TanninOne** [comment]
Please *do not* report multiple issues in one report! I will track only the first problem and close this report when that is done.

**Comment by Astor1977** [comment]
LOOT launches for me, just doesn't sort anything 90% of the time.  /shrug

**Comment by AreFinstad** [comment]
just gonna report that i got issues aswell. Error Code 1 when using the sort button and loot stops working when using MO as a launcher (worked with Skyrim a year or so ago).

note that im also using the newest snapshot from the nexus.

**Comment by Leer0x** [comment]
@AreFinstad Yes that is what I am getting, so it's good to see this is a sort of global error with any version of loot

**Comment by Leer0x** [comment]
@Astor1977 What version do you use? And do you use AntiVirus and elevate privileges for your programs? Loot doesn't launch AT ALL for me.

**Comment by Leer0x** [comment]
@TanninOne Will create reports for the other issues, I was really trying to figure out how I can solve the problems I was having.
Understood

**Comment by Astor1977** [comment]
@Leer0x I'm using LOOT  v0.8.1-323.  My AV is Bitdefender Free, and no, i don't elevate LOOT or MO.

I have LOOT installed OUT of my Program Files folder, and i have "--game=Fallout4" (no quotes) pasted in the Arguments box in the MO EXE Configuration options window for LOOT (hope that makes sense).

**Comment by Leer0x** [comment]
@Astor1977 Ok that is pretty much my similar setup. Hopefully Tannin can find the an error in the code or a workaround because I don't believe there is a fix, and is a problem with MO, I've never gotten Loot to run.

**Comment by TanninOne** [comment]
This report is becoming messy because different issues are being discussed (which is one reason you don't create a report about multiple things) so I'm going to delete all comments not related to nexus site login.

**Comment by TanninOne** [comment]
Please explain this: "MO does not properly login to Nexus Mods, it logs in correctly on download, but never manually"
There is no way to login manually in MO so I'm confused. MO logs into nexus when any operation requires an account, like downloading or endorsing. There is no "log in" button!

Also: What browser are you using to download? Have you tried downloading with a different one? Is it possible you installed MO in c:\program files or another "write restricted" location? Do you have a portable or regular installation?

**Comment by Leer0x** [comment]
It's not really a manual login, but I just mean typing the username and password in the "Nexus" settings tab.
The first thing I do after that is click the small mod settings' "Check for update" button and it gives me a "login failed" message. I press try again, and it asks me to put in my login info. Once again: Login failed.  Remember, I know my password is correct.
The weird thing here is, when I download a mod from Nexus, it doesn't say login failed, or anything, doesn't even say connecting to Nexus, it just downloads it, and says download started in the M.O. page.
What I THINK is that login fails every time the "Connecting to Nexus" messages pops up, and fails to login (And it does every time). However I can download files fine, just can't endorse or check for mod updates.

Now this may be confusing, because I said my nexus links didn't work.
But I fixed it (I THINK this is what fixed it) by logging out, then shutting down.
This morning I booted back up and everything worked fine Nexus link wise.

**Comment by Astor1977** [comment]
@Leer0x I have found that with this iteration of MO, restarting the program or the PC entirely rectifies various "quirks". At least for a while.  Sometimes, my game doesn't launch at all, or has mod issues, until i restart MO or the PC.  I don't know why, i can't even begin to guess, but with my OS on an SSD, rebooting takes less than a minute, so it's not a big deal.

**Comment by Leer0x** [comment]
@Astor1977 Makes sense. Glad to see it wasn't my fault, because the first time I restarted, the same thing happened, and I changed literally nothing. It wasn't until today that my problem fixed itself.

**Comment by GSDFan** [comment]
I believe your login issue is what we have seen in the forums on STEP. MO stores the info in the some .DAT files in the webcache folder.

Look at the paths tab to see where the webcache folder is located and then shut down MO. Use Windows file explorer and navigate to the webcache folder and look for cookies.dat and nexus_cookies.dat then archive them for safe keeping.

After you do this delete them and restart MO and try to use the function that is not working.

**Comment by Leer0x** [comment]
@GSDFan Thanks for the help.
I deleted the files, and restarted my PC to make sure, and it seems it is a global problem for login to fail after hitting "Endorse" or "Check for updates".

**Comment by L-a-n-g-o-l-i-e-r-s** [comment]
The issue appears to be because the Nexus is moving to SSL, can someone push an update to 1.3.11 and newer?? 

http://www.nexusmods.com/wasteland2/news/12780/?

Thanks.

**Comment by TanninOne** [comment]
@L-a-n-g-o-l-i-e-r-s this is not caused by the move to ssl, if it were you would see a connection error.
MO had a bug that passwords weren't escaped correctly, which means some special characters were not correctly sent to the nexus page. In particular the + sign was problematic, I don't know about others  (%, ?, ; and & are likely candidates but I haven't tested)

This should be solved in 2.0.6. If you're using an older version of MO you have to change your password to one that doesn't contain any of the problematic symbols (maybe this takes a bit of trial&error)
